code/modules/food_and_drinks/machinery/processor.dm

```diff
@@ -213,27 +218,26 @@
 				break
 	if(!picked_slime)
 		return
-	var/datum/food_processor_process/P = PROCESSOR_SELECT_RECIPE(picked_slime)
-	if (!P)
+	var/datum/food_processor_process/recipe = PROCESSOR_SELECT_RECIPE(picked_slime)
+	if (!recipe)
 		return
 
 	visible_message(span_notice("[picked_slime] is sucked into [src]."))
 	LAZYADD(processor_contents, picked_slime)
 	picked_slime.forceMove(src)
 
 /obj/machinery/processor/slime/process_food(datum/food_processor_process/recipe, atom/movable/what)
-	var/mob/living/simple_animal/slime/S = what
-	if (istype(S))
-		var/C = S.cores
-		if(S.stat != DEAD)
-			LAZYREMOVE(processor_contents, S)
-			S.forceMove(drop_location())
-			S.visible_message(span_notice("[C] crawls free of the processor!"))
+	var/mob/living/simple_animal/slime/processed_slime = what
+	if (istype(processed_slime))
+		var/core_count = processed_slime.cores
+		if(processed_slime.stat != DEAD)
+			processed_slime.forceMove(drop_location())
+			processed_slime.visible_message(span_notice("[core_count] crawls free of the processor!"))
 			return
-		for(var/i in 1 to (C+rating_amount-1))
-			var/atom/movable/item = new S.coretype(drop_location())
+		for(var/i in 1 to (core_count+rating_amount-1))
+			var/atom/movable/item = new processed_slime.coretype(drop_location())
 			adjust_item_drop_location(item)
-			SSblackbox.record_feedback("tally", "slime_core_harvested", 1, S.colour)
+			SSblackbox.record_feedback("tally", "slime_core_harvested", 1, processed_slime.colour)
 	..()
```

Comment:
```suggestion
	return ..()
```

Or maybe we should just delete this line altogether if we implement the early return on line 231? Investigate if this needs to be axed.

Reply:
Can't delete the call, as that is where the code that gibs the slime is found.

Reply:
I could move that bit of the code here, but that might be a problem in case someone adds a new recipe that uses mobs.